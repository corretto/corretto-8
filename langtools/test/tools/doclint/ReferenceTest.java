/*
 * @test /nodynamiccopyright/
 * @bug 8004832 8020556 8002154
 * @summary Add new doclint package
 * @build DocLintTester
 * @run main DocLintTester -Xmsgs:-reference ReferenceTest.java
 * @run main DocLintTester -ref ReferenceTest.out ReferenceTest.java
 */

/** */
public class ReferenceTest {
    /**
     * @param x description
     */
    public int invalid_param;

    /**
     * @param x description
     */
    public class InvalidParam { }

    /**
     * @param x description
     */
    public void param_name_not_found(int a) { }

    /**
     * @param <X> description
     */
    public class typaram_name_not_found { }

    /**
     * @see Object#tooStrong()
     */
    public void ref_not_found() { }

    /**
     * @return x description
     */
    public int invalid_return;

    /**
     * @return x description
     */
    public void invalid_return();

    /**
     * @throws Exception description
     */
    public void exception_not_thrown() { }

    /**
     * @param <T> throwable
     * @throws T description
     */
    public <T extends Throwable> void valid_throws_generic() throws T { }

    /**
     * {@link java.util.List<String>}
     * {@link java.util.List<String>#equals}
     * @see java.util.List<String>
     * @see java.util.List<String>#equals
     */
    public void invalid_type_args() { }
}

