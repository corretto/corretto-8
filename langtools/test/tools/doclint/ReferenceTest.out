ReferenceTest.java:13: error: invalid use of @param
     * @param x description
       ^
ReferenceTest.java:18: error: invalid use of @param
     * @param x description
       ^
ReferenceTest.java:23: error: @param name not found
     * @param x description
              ^
ReferenceTest.java:25: warning: no @param for a
    public void param_name_not_found(int a) { }
                ^
ReferenceTest.java:28: error: @param name not found
     * @param <X> description
               ^
ReferenceTest.java:33: error: reference not found
     * @see Object#tooStrong()
            ^
ReferenceTest.java:38: error: invalid use of @return
     * @return x description
       ^
ReferenceTest.java:43: error: invalid use of @return
     * @return x description
       ^
ReferenceTest.java:48: error: exception not thrown: java.lang.Exception
     * @throws Exception description
               ^
ReferenceTest.java:59: error: type arguments not allowed here
     * {@link java.util.List<String>}
              ^
ReferenceTest.java:60: error: type arguments not allowed here
     * {@link java.util.List<String>#equals}
              ^
ReferenceTest.java:61: error: type arguments not allowed here
     * @see java.util.List<String>
            ^
ReferenceTest.java:62: error: type arguments not allowed here
     * @see java.util.List<String>#equals
            ^
12 errors
1 warning

